### PR TITLE
- changes to allow for multi-uav support

### PR DESCRIPTION
**Pull request by blackcoder**
Modifications to allow simulating multi-UAVs, the following modifications were made:

  - Mavlink class can now take the mavlink_fcu_url as a ros parameter
  - ardrone and iris launch files expect this argument to be provided, otherwise they will default to: udp://localhost:14565@localhost:14560
  - A new multi_uav.launch file was added to demonstrate this capability

### CODE REVIEW

**Comment by mzahana** [comment]
@TarekTaha This is a great feature. However, I noticed that this requires ```rotors_simulator``` package. After I built ```rotors_simulator``` from source, and run ```roslaunch px4 multi_uav.launch```, I get the following error:

```
while processing /home/mzahana/ros_ws/src/rotors_simulator/rotors_gazebo/launch/spawn_iris.launch:
Invalid roslaunch XML syntax: [Errno 2] No such file or directory: u'/home/mzahana/ros_ws/src/rotors_simulator/rotors_gazebo/launch/spawn_iris.launch'
The traceback for the exception was written to the log file
```

So, I looked into the ```rotors_gazebo/launch/``` folder, but I did not find the ```spawn_iris.launch``` file.

Also, if you check the ```rotors_gazebo``` [repository](https://github.com/ethz-asl/rotors_simulator/tree/master/rotors_gazebo/launch) , you won't find  ```spawn_ardrone.launch``` or ```spawn_iris.launch```

So, I guess that is why it does not work.

Does the launch files of the models have to be from ```rotors_gazebo```? I guess it is better to default to models inside the ```px4``` package to make sure all is self-contained, no?
Thanks.

**Comment by TarekTaha** [comment]
@mzahana This multi-uav support does not work anymore due to various changes in the Firmware. 

In preparation for MBZIRC, we decided to shift to sitl_gazebo and we had to fiddle a bit with the settings in order to enable multi-uav support. 

Please check this repository : https://github.com/kuri-kustar/kuri_mbzirc_sim and in particular : https://github.com/kuri-kustar/kuri_mbzirc_sim/blob/master/launch/mavros_posix_sitl_px4_ch3.launch https://github.com/kuri-kustar/kuri_mbzirc_sim/tree/master/config https://github.com/kuri-kustar/kuri_mbzirc_sim/tree/master/urdf/iris 
Basically, we had to modify the way the Firmware talks to the sitl_gazebo, and modify the ports used to talk to each drone.
Let me know if you need any help.

**Comment by mzahana** [comment]
@TarekTaha  Thanks for the explanation. I think I see the complete picture now and I know what to modify/setup.

Basically,
* each vehicle has a seperate sdf model that includes a unique mavlink udp port for gazebo to communicate with px4 app
* startup file ```rsC``` for each vehicle, where a mavlink instance is srarted at the same port in the previous step. Another mavlink instance can be started on a unique port to communicate with mavros
* a launch file that runs gazebo with a world model, loads each vehicle (spawn),  runs ```px4``` app with corresponding rsC startup file for each vehicle, runs mavros for each vehicle.

I will do some testing and issue a PR. I will also do a documentation for the steps as there are no clear documentation for that feature. 

Thanks and let me know if you have comments.

**Comment by TarekTaha** [comment]
@mzahana correct. It's not the most elegant or general solution, but it works. It would be great if you can generalize the solution so that all the ports are picked up from the launch file or yaml file, and spawns multiple instances of the UAV model with the new port. 

Just one thing to take care of, we had to play with a lot of port numbers to find ones that work.

**Comment by mzahana** [comment]
I created PR #7235 which tries to generalize the approach as much as possible. Noe need for multiple SDF for each vehicle.

This PR https://github.com/PX4/sitl_gazebo/pull/101 is required too.

Feedback is appreciated.
